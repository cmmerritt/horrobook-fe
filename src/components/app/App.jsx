import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Form from '../forms/Form';
import FavoriteList from '../favorites/FavoriteList';

export default function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Form} />
        <Route exact path="/favorites" component={FavoriteList} />
      </Switch>
    </Router>
  );
}
