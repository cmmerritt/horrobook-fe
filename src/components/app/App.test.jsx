import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<Router><App /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
