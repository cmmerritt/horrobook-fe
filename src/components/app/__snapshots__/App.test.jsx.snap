// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <section>
    <h2>
      Add a new favorite book
    </h2>
    <section>
      <form>
        <label>
          Title:
          <input
            name="title"
            required=""
            value=""
          />
        </label>
        <label>
          Author:
          <input
            name="author"
            required=""
            value=""
          />
        </label>
        <div>
          <button>
            Add to favorites
          </button>
        </div>
      </form>
    </section>
  </section>
</DocumentFragment>
`;
